wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/remote/LsConfigurationDetector.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver.remote;
+
+import java.util.Map;
+import javax.inject.Singleton;
+
+/** Detects if machine server attributes indicates that we are dealing with language server. */
+@Singleton
+class LsConfigurationDetector {
+  /**
+   * Tests attributes for a language server indicator
+   *
+   * @param attributes map with machine server attributes
+   * @return true if language server is detected, false otherwise
+   */
+  boolean isDetected(Map<String, String> attributes) {
+    return "ls".equals(attributes.get("type"));
```

Comment:
Do we have this "magic" constant described anywhere in our javadocs?

Reply:
No, we don't have it, I'd say it's an internal implementation and it is not as important as, for example, the fact that this constant is not described in the documentation, because as far as I know the documentation is under construction at the moment, so I guess we'll have to wait a bit until it is ready. Could you satisfy my curiosity, tell me please why are you taking into account "ls" constant and not "type" that stands next, or others, is it somehow special for you?

Reply:
Just pointed out that value. And yes, just `type` does look a bit simple for me since attributes might contain different attributes for different ports of the application. But this is not a question to you, personally, or this PR since I know that it was discussed by maintainers.